RPG Wizardry/Assets/Scripts/GameWorld/Room.cs

```diff
@@ -5,12 +5,24 @@ namespace nl.SWEG.RPGWizardry.GameWorld
 {
     public class Room : MonoBehaviour
     {
+        #region Inner Types
+        /// <summary>
+        /// Creating Array for event that checks for Room Clear
+        /// </summary>
+        public delegate void roomClear();
+
+        #endregion
         #region Variables
         #region Public
         /// <summary>
         /// Whether the Room has been Cleared (No more Enemies)
         /// </summary>
         public bool Cleared { get; private set; }
+
+        /// <summary>
+        /// Roomcleared event
+        /// </summary>
+        public static roomClear clearedRoom;

```

Comment:
A cleaner way to do this:
- Make this event Non-Static.
- Grab references to the Current Room when needed through the FloorManager
- Create an Event in FloorManager that runs whenever a new Room is loaded, so that you can subscribe to that if you need to check future rooms when those are loaded.